paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/MergeIntoPaimonTable.scala

```diff
@@ -150,7 +150,7 @@ case class MergeIntoPaimonTable(
           case _ => false
         }
       }
-      if (hasUpdate(matchedActions)) {
+      if (hasUpdate(matchedActions) || notMatchedActions.nonEmpty) {
```

Comment:
Can you explain this line in detail?

Reply:
I find this line is not necessary, revert this

Reply:
when there is matched or  not matched action by target, we should evaluate on the file splits after inner join

Reply:
Hi @YannByron , I have another thought for this question, and opened another [PR](https://github.com/apache/paimon/pull/5195)

I think we should not use this condition to move this untouched files into touched file set, which will increase the rewrite input/output size. 

Instead, if not matched action is non empty, these files should be input as untouched files, and we only have to consume the untouched files only when the not matched by target action is present